Userland/Libraries/LibGUI/TextEditor.cpp

```diff
@@ -2512,4 +2568,46 @@ void TextEditor::highlighter_did_set_folding_regions(Vector<GUI::TextDocumentFol
     recompute_all_visual_lines();
 }
 
+ErrorOr<TextEditor::GutterIndicatorID> TextEditor::register_gutter_indicator(PaintGutterIndicator draw_indicator, OnGutterIndicatorClick on_click)
```

Comment:
This only returns errors if we fail to append to `m_gutter_indicators`. Since we know we only have 32 at most, we could simply ensure that capacity in the constructor and then wouldn't need to propagate an error here. But, that means allocating some extra memory we don't usually need.